src/capturer/mongo/__tests__/utils.test.ts

```diff
@@ -0,0 +1,333 @@
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+import {
+  Collection,
+  CollectionInfo,
+  Db,
+  InsertOneResult,
+  ListCollectionsCursor,
+  MongoClient,
+} from "mongodb";
+import {
+  disconnectMongo,
+  findDocumentByID,
+  getMongoCollection,
+  getMongoDb,
+  getOrCreateMongoCollection,
+  insertDocument,
+} from "../utils";
+
+jest.mock("mongodb", () => ({
+  MongoClient: jest.fn(),
+}));
+
+const mockClient: MongoClient = {
+  connect: jest.fn(),
+  close: jest.fn(),
+  db: jest.fn(),
+} as unknown as MongoClient;
+
+const mockDb: Db = {
+  collection: jest.fn(),
+  listCollections: jest.fn(),
+  createCollection: jest.fn(),
+} as unknown as Db;
+
+describe("getMongoDb", () => {
+  it("should get MongoDB database", async () => {
+    const dbSpy = jest
+      .spyOn(mockClient, "db")
+      .mockReturnValueOnce(mockDb as unknown as Db);
+
+    const result = getMongoDb(mockClient, "mock-db");
+
+    expect(result).toEqual(E.right(mockDb));
+    expect(dbSpy).toHaveBeenCalledWith("mock-db");
+  });
+
+  const getDBError = new Error("Error while getting database");
+  it("should handle error when getting database", async () => {
+    const dbSpy = jest.spyOn(mockClient, "db").mockImplementation(() => {
+      throw getDBError;
+    });
+
+    const result = getMongoDb(mockClient, "mock-db");
+
+    expect(result).toEqual(
+      E.left(
+        new Error(
+          "Impossible to Get the mock-db db: Error: Error while getting database"
+        )
+      )
+    );
+    expect(dbSpy).toHaveBeenCalledWith("mock-db");
+  });
+});
+
+const mockCollection: Collection = {
+  find: jest.fn(),
+  sort: jest.fn(),
+  limit: jest.fn(),
+  tryNext: jest.fn(),
+  findOne: jest.fn(),
+  insertOne: jest.fn(),
+} as unknown as Collection;
+
+const collectionName = "testCollection";
+const mockToArray: CollectionInfo[] = [{ name: collectionName }];
+const mockListCollectionsCursor: ListCollectionsCursor = {
+  toArray: jest.fn,
+} as unknown as ListCollectionsCursor;
+
+describe("getMongoCollection", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+  it("should get MongoDB collection", async () => {
+    jest
+      .spyOn(mockDb, "listCollections")
+      .mockReturnValueOnce(mockListCollectionsCursor);
+    jest
+      .spyOn(mockListCollectionsCursor, "toArray")
+      .mockReturnValueOnce(Promise.resolve(mockToArray));
```

Comment:
we can move spyOn implementation outside single test an mock each implementation singularly :)